src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+            {
+                ContinuousProperty& continuousProp =
+                    static_cast< ContinuousProperty& >( property );
+
+                if( ( property.getValuesHdfDatatype()
+                            == AbstractValuesProperty::FLOAT
+                        || property.getValuesHdfDatatype()
+                               == AbstractValuesProperty::DOUBLE ) )
+                {
+                    GEO::Attribute< double > attribute;
+                    attribute.create_vector_attribute(
+                        attri_manager, property.getTitle(), el_per_val );
+
+                    std::unique_ptr< double[] > values( new double[val_count] );
+                    continuousProp.getDoubleValuesOfPatch(
+                        patch_index, &values[0] );
+                    for( auto val : range( val_count ) )
+                    {
+                        attribute[val] = values[val];
+                    }
+                }
+            }
+            else if( dynamic_cast< DiscreteProperty* >( &property ) != nullptr )
+            {
+                DiscreteProperty* discreteProp =
+                    dynamic_cast< DiscreteProperty* >( &property );
+
+                GEO::Attribute< int > attribute;
+                attribute.create_vector_attribute(
+                    attri_manager, property.getTitle(), el_per_val );
+
+                std::unique_ptr< int[] > values( new int[val_count] );
+                discreteProp->getIntValuesOfPatch( patch_index, &values[0] );
+                for( auto val : range( val_count ) )
+                {
+                    attribute[val] = values[val];
+                }
+            }
+
+            return true;
+        }
+    } // namespace
+
+    bool GeoModelBuilderRESQMLImpl::read_surfaces( const EpcDocument& pck )
+    {
+        std::vector< TriangulatedSetRepresentation* > all_tri_set_rep =
+            pck.getAllTriangulatedSetRepSet();
+
+        if( all_tri_set_rep.empty() )
+        {
+            throw RINGMeshException(
+                "RESQML2", "At least one surface is required" );
+        }
+
+        for( auto tri_set : all_tri_set_rep )
+        {
+            showAllMetadata( tri_set );
+
+            const gmge_id interface_id =
+                builder_.geology.create_geological_entity(
+                    Interface3D::type_name_static() );
+
+            AbstractFeature* feature = nullptr;
+            if( tri_set->getInterpretation() != nullptr )
+            {
+                feature = tri_set->getInterpretation()->getInterpretedFeature();
+            }
+
+            if( feature != nullptr )
+            {
+                const Interface3D& interface =
+                    static_cast< const Interface3D& >(
+                        geomodel_.geological_entity(
+                            GeologicalEntityType( "Interface" ),
+                            interface_id.index() ) );
+                const_cast< Interface3D& >( interface )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor the code to not need "const_cast". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS859)